### ISSUE TITLE
Object-Oriented Refactor

### ISSUE THREAD

**Issue by zenflow** [sweetalert2/sweetalert2#1007]
Related to this comment https://github.com/sweetalert2/sweetalert2/issues/970#issue-301393086  and this comment https://github.com/sweetalert2/sweetalert2/issues/815#issuecomment-357433667 (where I was wrong about sweetalert2 becoming OO).. it is possible to have a proper Object-Oriented implementation (with a bit of breaking change, but so simple that a codemod could take care of 80% of the work in most cases)

We can skip the `SwalContext` I mentioned in #970 and go right exporting a proper class...

```js
// before
const { value: daddy } = await swal({ input: 'text', title: 'Who\'s your daddy?' })
// after
const { value: daddy } = await new Swal({ input: 'text', title: 'Who\'s your daddy?' })

// before
const { value } = await swal({
  input: 'text', 
  onOpen: () => {
    setTimeout(() => swal.clickConfirm(), 1000)
  }
})
// after (boring)
const { value } = await new Swal({
  input: 'text', 
  onOpen: swal => { // `swal` is the instance of `Swal`
    setTimeout(() => swal.clickConfirm(), 1000)
  }  
})
// or (here's the cool part.. I think this reads nicer.. onOpen is still good for mixins[1] though)
const swal = new Swal({ input: 'text' })
await swal.opened
setTimeout(() => swal.clickConfirm(), 1000)
const { value } = await swal
```

[1] https://github.com/sweetalert2/sweetalert2/issues/968#issuecomment-372039240 

In theory the change would be quite straight-forward to implement, and isolated from merge conflicts, **once #970 is resolved**.

**Comment by zenflow** [comment]
Gimme your feels y'all @limonte @toverux @acupajoe @samturrell

**Comment by zenflow** [comment]
This issue seems to be blocking (in my mind).. https://github.com/sweetalert2/sweetalert2/pull/1008#issuecomment-372662854

**Comment by zenflow** [comment]
Let me just do a bit on #970 and then I can put in a PR to show more of what this would look like.

**Comment by zenflow** [comment]
The PR is in.. #1036

**Comment by zenflow** [comment]
Closed by #1036
